util/flask_util.py

```diff
@@ -92,18 +92,28 @@ def _headers(self, headers={}):
         # Don't modify the underlying dictionary; it came from somewhere else.
         headers = dict(headers)
 
-        # Set Cache-Control based on max-age.
+        # Set headers based on privacy settings and maximum age.
         if self.private:
             private = "private"
+
+            # A private resource should be re-requested, rather than
+            # retrieved from cache, if the authorization credentials
+            # change from those originally used to retrieve it.
+            headers['Vary'] = 'Authorization'
         else:
             private = "public"
         if self.max_age and isinstance(self.max_age, int):
-            # A CDN should hold on to the cached representation only half
-            # as long as the end-user.
             client_cache = self.max_age
-            cdn_cache = self.max_age / 2
-            cache_control = "%s, no-transform, max-age=%d, s-maxage=%d" % (
-                private, client_cache, cdn_cache
+            if self.private:
+                # A private resource should not be cached by
+                # intermediaries at all.
+                s_maxage = ""
+            else:
+                # A public resource can be cached by intermediaries
+                # for half as long as the end-user can cache it.
+                s_maxage = ", s-maxage=%d" % (self.max_age / 2)
+            cache_control = "%s, no-transform, max-age=%d%s" % (
+                private, client_cache, s_maxage
```

Comment:
I might be following this logic incorrectly, but just to be certain, when `self.private == True`, the `cache-control` header should have `private`, and when `self.private == False` then `cache-control` doesn't need the `private` directive.

Reply:
Yes, the code starting here sets `private` to either "private" or "public" depending on the value `self.private`:

https://github.com/NYPL-Simplified/server_core/pull/1258/files/075647968fee78d86d21931fe233b9420593a56f#diff-40292c475f9d61e929ebbc06204673b0d7e5e58acf523ce1b00d18b17ae6f089R95

So `private` could be more accurately called `private_status` or something.